MirrorProvider/MirrorProvider.Linux/LinuxFileSystemVirtualizer.cs

```diff
@@ -103,14 +104,14 @@ public override void Stop()
                     }
                     else
                     {
-                        // The MirrorProvider marks every file as executable (mode 755), but this is just a shortcut to avoid the pain of
-                        // having to p/invoke to determine if the original file is exectuable or not.
-                        // A real provider will have to get this information from its data source. For example, GVFS gets this info
-                        // out of the git index along with all the other info for projecting files.
-                        UInt16 fileMode = Convert.ToUInt16("755", 8);
+                        string combinedPath = Path.Combine(relativePath, child.Name);
+                        int r = Syscall.lstat(this.GetFullPathInMirror(combinedPath), out Stat stat);
+                        if (r == -1)
+                            return Result.EIOError;
```

Comment:
Coding conventions for VFS for Git (documentation coming Real Soon :)) dictate always bracing.

Reply:
👍